File: MPPCS/Repository/Repository.cs
using CommonDomain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Persistence.Repository
{
    public interface Repository<E> where E : Entity<int>
    {
        E Find(int id);
        List<E> GetAll();
        void Add(E entity);
        E Delete(int id);
        E Update(E entity);
        int Size();
    }
}

File: PersistenceAndServices/src/main/java/repository/Repository.java
package repository;

import domain.Entity;
import java.util.List;


public interface Repository<ID, E extends Entity<ID>> {
    E find(ID id) throws Exception;
    List<E> getAll() throws Exception;
    void add(E entity) throws Exception;
    E delete(ID id) throws Exception;
    E update(E entity) throws Exception;
    Integer size() throws Exception;

}